src/test/java/build/buildfarm/ByteStreamServiceTest.java

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buildfarm;
+
+import build.buildfarm.common.Digests;
+import build.buildfarm.server.BuildFarmInstances;
+import build.buildfarm.server.ByteStreamService;
+import com.google.bytestream.ByteStreamProto.WriteRequest;
+import com.google.bytestream.ByteStreamProto.WriteResponse;
+import com.google.devtools.remoteexecution.v1test.Digest;
+import com.google.protobuf.ByteString;
+import io.grpc.stub.StreamObserver;
+import java.util.Random;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ByteStreamServiceTest {
+
+  private static byte[] generateFileData(Random random) {
```

Comment:
I'd pass in the length of the data to randomly generate.

Reply:
Agreed, and this also warrants a squash with the other commit for the addition of this test class.

Reply:
> I'd pass in the length of the data to randomly generate.

Long done.

Want to get the format hook in before continuing. Depends on upstream google-java-format so might take a while.